src/components/cl/hier/cl_hier_team.c

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "cl_hier.h"
+#include "utils/ucc_malloc.h"
+#include "core/ucc_team.h"
+#include "core/ucc_service_coll.h"
+
+#define SBGP_SET(_team, _sbgp, _enable)                                        \
+    _team->sbgps[UCC_HIER_SBGP_##_sbgp].sbgp_type = UCC_SBGP_##_sbgp;          \
+    _team->sbgps[UCC_HIER_SBGP_##_sbgp].state     = UCC_HIER_SBGP_##_enable;
+
+/* The function below must enable/disable those hier sbgps that will be
+   used to construct hierarchical schedules.
+   Currently just enable two sbgps as example and for testing purposes.
+   Next step is to enable sbgps based on the requested hierarchical algs. */
+static void ucc_cl_hier_enable_sbgps(ucc_cl_hier_team_t *team)
+{
+    SBGP_SET(team, NODE, ENABLED);
+    SBGP_SET(team, NET, ENABLED);
+}
+
```

Comment:
Do we expect two CLs to be enabled at the same time? Say, I want to do one collective with hierarchical team and another with basic team?